dispatch.go

```diff
@@ -89,22 +90,35 @@ func newDispatch(enc encoder, dec byteReadingDecoder, l LogInterface, wef WrapEr
 
 type call struct {
 	ch             chan error
+	doneCh         chan struct{}
 	method         string
 	seqid          int
 	arg            interface{}
 	res            interface{}
 	errorUnwrapper ErrorUnwrapper
 	profiler       Profiler
+	ctx            context.Context
 }
 
-func newCall(m string, arg interface{}, res interface{}, u ErrorUnwrapper, p Profiler) *call {
+func newCall(ctx context.Context, m string, arg interface{}, res interface{}, u ErrorUnwrapper, p Profiler) *call {
 	return &call{
-		ch:             make(chan error),
+		ch:             make(chan error, 1),
```

Comment:
Worth a comment to say that you're buffering it so that Done will always succeed the first time.